### PR TITLE
many: implement snapd API install system finish step

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**

### CODE REVIEW

File: overlord/devicestate/handlers_install.go [hunk 1]
```diff
@@ -1214,14 +1262,115 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
...
+	logger.Debugf("making the recovery partition bootable")
+	if err := boot.MakeBootableImage(sys.Model, seedDir, bootWith, boot.ModeRun, []string{}); err != nil {
+		return err
+	}
+
+	// sets environment
+	logger.Debugf("making the recovery partition runnable")
+	if err := boot.MakeRecoverySystemBootable(seedDir, "/", recovBootWith); err != nil {
+		return err
+	}
```
Comment (pedronis) [inline overlord/devicestate/handlers_install.go:1361]: the feel like abuse, especially the second one passing "/" as we have no recovery systems and also I think it writes some files that we will then just ignore?.  It seems we need a different helper that does the part of these two that we really

File: overlord/devicestate/handlers_install.go [hunk 2]
```diff
@@ -1214,14 +1262,115 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
...
+		Gadget:     snapInfos[snap.TypeGadget],
+		GadgetPath: snapSeeds[snap.TypeGadget].Path,
+
+		UnpackedGadgetDir: mntPtForType[snap.TypeGadget],
+
+		RecoverySystemLabel: systemLabel,
+		// TODO: important but a bit unclear why needed
+		RecoverySystemDir: systemLabel,
+
+		Recovery: true,
```
Comment (pedronis) [inline overlord/devicestate/handlers_install.go:1342]: this looks wrong and at odd with the change done in MakeRunnableSystem

File: overlord/devicestate/handlers_install.go [hunk 3]
```diff
@@ -1214,14 +1262,115 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
...
+
+		// XXX: rename to something more neutral like "TargetDir" ?
+		InstallHostWritableDir: filepath.Join(boot.InitramfsRunMntDir, "ubuntu-data"),
```
Comment (pedronis) [inline overlord/devicestate/handlers_install.go:1345]: it feels like we need something more like MakeRunneableSystemUnder or ...WithTarget

**Review by pedronis — COMMENTED** [review]
some quick comments

File: overlord/devicestate/handlers_install.go [hunk 4]
```diff
@@ -1214,14 +1262,115 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
...
+		kernelInfo.Revision = snap.R(-1)
+	}
+
+	// TODO useEncryption equals true case
+	useEncryption := false
+	installObserver, trustedInstallObserver, err := buildInstallObserver(sys.Model, mntPtForType[snap.TypeGadget], useEncryption)
+	if err != nil {
+		return err
+	}
+
```
Comment (pedronis) [inline overlord/devicestate/handlers_install.go:1311]: this needs to mount the partitions like install.Run would, as that is not a task for the installer as such

**Comment by alfonsosanchezbeato** [comment]
Closing, as the relevant parts in which this one was splitted are now merged.

**Pull request closed without merging**